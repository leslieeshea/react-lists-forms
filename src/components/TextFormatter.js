import React, { PureComponent } from 'react';
import figlet from 'figlet';

export default class TextFormatter extends PureComponent {
  state = {
    text: '',
    formattedText: '',
    color: 'black',
    font: 'Ghost'
  };

  formatText = () => {
    const { text, font } = this.state;
    figlet.text(text, {
      font: font
    }, (err, result) => {
      this.setState({ formattedText: result });
    });
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value }, () => {
      this.formatText();
    });
  }

  render() {
    const { text, formattedText, color, font } = this.state;

    const fontOptions = ['Ghost', 'Stacey', 'Bear', 'Tinker-Toy'].map(font => {
      return <option key={font} value={font}>{font}</option>;
    });

    return (
      <>
        <select name="font" value={font} onChange={this.handleChange}>{fontOptions}</select>
        <input name="text" value={text} onChange={this.handleChange} />
        <input name="color" type="color" value={color} onChange={this.handleChange} />
        <pre style={{ color }}>{formattedText}</pre>
      </>
    );
  }
}
