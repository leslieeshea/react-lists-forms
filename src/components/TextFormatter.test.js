import React from 'react';
import { shallow } from 'enzyme';
import TextFormatter from './TextFormatter';

describe('TextFormatter component', () => {
  it('renders the TextFormatter', () => {
    const wrapper = shallow(<TextFormatter />);
    expect(wrapper).toMatchSnapshot();
  });
});
