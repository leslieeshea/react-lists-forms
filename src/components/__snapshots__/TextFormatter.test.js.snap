// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFormatter component renders the TextFormatter 1`] = `
<Fragment>
  <select
    name="font"
    onChange={[Function]}
    value="Ghost"
  >
    <option
      key="Ghost"
      value="Ghost"
    >
      Ghost
    </option>
    <option
      key="Stacey"
      value="Stacey"
    >
      Stacey
    </option>
    <option
      key="Bear"
      value="Bear"
    >
      Bear
    </option>
    <option
      key="Tinker-Toy"
      value="Tinker-Toy"
    >
      Tinker-Toy
    </option>
  </select>
  <input
    name="text"
    onChange={[Function]}
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="black"
  />
  <pre
    style={
      Object {
        "color": "black",
      }
    }
  />
</Fragment>
`;
